operators/endpointmetrics/pkg/util/status.go

```diff
@@ -34,7 +34,10 @@ var (
 	}
 )
 
-func ReportStatus(ctx context.Context, client client.Client, i *oav1beta1.ObservabilityAddon, t string) {
+func ReportStatus(ctx context.Context, client client.Client, i *oav1beta1.ObservabilityAddon, t string, reportStatus bool) {
+	if !reportStatus {
+		return
+	}
```

Comment:
It seems very weird to have a function that takes a boolean that makes such function do nothing. I think this check should be done by the callers of `ReportStatus` before they decide wether to call it.

Reply:
it was previously implemented that way where we would check if we needed to call  but report status was called in multiple places and in a previous review comment was asked to check for obsaddon and report status accordingly